_posts/2023-03-27-group-equivariant-cnn-2/group-equivariant-cnn-2.Rmd

```diff
@@ -0,0 +1,374 @@
+---
+title: "Implementing rotation equivariance: Group-equivariant CNN from scratch"
+description: >
+  We code up a simple group-equivariant convolutional neural network (GCNN) that is equivariant to rotation. The world may be upside down, but the network will know.
+author:
+  - name: Sigrid Keydana
+    affiliation: Posit
+    affiliation_url: https://www.posit.co/
+slug: keydanagcnn1
+date: 2023-03-27
+categories:
+  - Torch
+  - R
+  - Spatial Data
+  - Image Recognition & Image Processing
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/preview.jpg
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE, fig.width = 6, fig.height = 6)
+```
+
+Convolutional neural networks (CNNs) are great -- they're able to detect features in an image no matter where. Well, not exactly. They're not indifferent to just any kind of movement. Shifting up or down, or left or right, is fine; rotating around an axis is not. That's because of how convolution works: traverse by row, then traverse by column (or the other way round). If we want "more" (e.g., successful detection of an upside-down object), we need to extend convolution to an operation that is *rotation-equivariant*. An operation that is *equivariant* to some type of action will not just register the moved feature per se, but also, keep track of which concrete action made it appear where it is.
```

Comment:
> they're able to detect features in an image no matter where. Well, not exactly. They're not indifferent to just any kind of movement.

Do you mean where in the image the features are? I think this is what you mean, but I got a little thrown by your use of the word "movement" in the following sentence, since it made me think we were looking at something non-static. 

You might not need to change anything, just wanted to make sure I was following correctly.